byterun/finalise.c

```diff
@@ -97,13 +104,26 @@ void caml_final_update (void)
     young = j;
     to_do_tl->size = k;
     for (i = 0; i < k; i++){
-      /* Note that item may already be dark due to multiple entries in
-         the final table. */
-      caml_darken (to_do_tl->item[i].val, NULL);
+      if(!(final_table[i].flags & FINAL_CALLED_WITHOUT_VALUE)) {
```

Comment:
Isn't this condition reversed? If you call without a value, you do **not** want to darken, right?